csharp/extractor/Semmle.Extraction.CSharp/SymbolExtensions.cs

```diff
@@ -148,29 +151,19 @@ void WriteType(Context cx, TextWriter trapFile, ITypeSymbol symbol, ISymbol symb
         /// </summary>
         /// <param name="cx">The extraction context.</param>
         /// <param name="trapFile">The trap builder used to store the result.</param>
-        /// <param name="prefix">Whether to prefix the type ID with the assembly ID.</param>
         /// <param name="symbolBeingDefined">The outer symbol being defined (to avoid recursive ids).</param>
-        /// <param name="subTermAction">The action to apply to syntactic sub terms of this type.</param>
-        public static void BuildTypeId(this ITypeSymbol type, Context cx, TextWriter trapFile, bool prefix, ISymbol symbolBeingDefined, Action<Context, TextWriter, ITypeSymbol, ISymbol> subTermAction)
-        {
-            if (type.SpecialType != SpecialType.None && !(type is INamedTypeSymbol n && n.IsGenericType))
-            {
-                /*
-                 * Use the keyword ("int" etc) for the built-in types.
-                 * This makes the IDs shorter and means that all built-in types map to
-                 * the same entities (even when using multiple versions of mscorlib).
-                 */
-                trapFile.Write(type.ToDisplayString());
-                return;
-            }
+        public static void BuildTypeId(this ITypeSymbol type, Context cx, TextWriter trapFile, ISymbol symbolBeingDefined) =>
```

Comment:
Base classes are only added to the IDs when they are non-`object`. But yes, if a class has two different base classes in two different compilations, then that will result in two different entities. And IMO that is better than getting one entity with two base classes (both because it violates a DB constraint, and because I don't think our various analysis will deal well with such entities).

Reply:
I think this is something that requires a bit more investigation and I've opened https://github.com/github/codeql-csharp-team/issues/70 to deal with this at a later point. Extracting them with a different ID makes sense for now.